bin/rad-diff

```diff
@@ -0,0 +1,350 @@
+                   (def diff-number (match selected {:diff {:number 'n}} n))
+                   (def pretty-item
+                     (add-comment-md
+                       (pretty-diff (lookup :diff selected))))
+                   (def edited-item (edit-in-editor! pretty-item))
+                   (def added-comment
+                     (string/unlines
+                      (take-while (fn [x] (not (eq? x comment-divider)))
+                                  (string/lines edited-item))))
+                   (put-str!
+                    (if (eq? added-comment "")
+                      "Item was not updated."
+                      (do (simple-add-comment! chain-ref diff-number added-comment get-git-username!)
+                          (string-append "Sent comment: " added-comment)))))
+               _ (put-str! "No selection made."))
+         (map (fn [d] (put-str! (first d))) items)))))
+
+(def show-diff!
+  (fn [chain-name n]
+    (if (number? n)
+      (if (integral? n)
+        (do (def chain-ref (make-chain-ref chain-name))
+          (def diff (verify-diff-number n chain-ref))
+          (put-str! (pretty-diff diff)))
+        (cmd-parse-failure "Diff numbers must be whole numbers!"))
+      (cmd-parse-failure "To show a diff, please provide the diff number."))))
+
+(def new-diff-chain!
+  "Create a new DIFF chain with the given URL"
+  (fn [chain-name]
+    (create-diffs-chain! (make-name chain-name))))
+
+(def edit-diff-state!
+  (fn [chain-name n new-state]
+    (if (number? n)
+      (if (integral? n)
+        (do (def chain-ref (make-chain-ref chain-name))
+          (verify-diff-number n chain-ref)
+          (simple-edit-diff! chain-ref n {:state new-state})
+          (if (eq? new-state :accepted)
+            (do
+              (push-diff! chain-ref n)
+              (put-str! (string-append "Diff " (show n) " has been " (pretty-state new-state) " and merged.")))
+            (put-str! (string-append "Diff " (show n) " has been " (pretty-state new-state)))))
+        (cmd-parse-failure "Diff numbers must be whole numbers!"))
+    (cmd-parse-failure "To change the state, please provide the diff number."))))
+
+(def create-comment!
+  (fn [chain-name n comment]
+    (if (number? n)
+      (if (integral? n)
+        (do (def chain-ref (make-chain-ref chain-name))
+          (verify-diff-number n chain-ref)
+          (simple-add-comment! chain-ref n comment get-git-username!)
+          (put-str! (string-append "Added comment to Diff " (show n))))
+        (cmd-parse-failure "Diff numbers must be whole numbers!"))
+    (cmd-parse-failure "To create a comment, please provide the diff number."))))
+
+(def valid-list-options
+  ["--new" "--pending" "--retracted" "--accepted" "--rejected" "--fancy"])
+
+(def list-options
+  (fn [os]
+    (map (fn [o]
+             (if (member? o valid-list-options)
+               (do (match (drop 2 o)
+                 "new"        :new
+                 "accepted"   :accepted
```

Comment:
There's a lot of repetition here, and opportunities for mismatch. You can just `read` the list option as a keyword (by prepending `:`) (and there are probably other ways too).